src/smc-webapp/project/history/search.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from "react";
+import { useDebounce } from "../../hooks";
+import { SearchInput } from "../../r_misc";
+import { ProjectActions } from "smc-webapp/project_store";
+import { EventRecordMap } from "./types";
+
+interface Props {
+  search?: string;
+  actions: ProjectActions;
+  selected?: EventRecordMap;
+  increment_cursor: () => void;
+  decrement_cursor: () => void;
+  reset_cursor: () => void;
+}
+
+export function LogSearch({
+  search,
+  selected,
+  actions,
+  reset_cursor,
+  increment_cursor,
+  decrement_cursor
+}: Props): JSX.Element {
+  const open_selected = React.useCallback(
+    (_value, info: any): void => {
+      const e = selected?.get("event");
+      if (e == undefined || typeof e === "string") {
+        return;
+      }
+
+      switch (e.get("event")) {
+        case "open":
+          const target = e.get("filename");
+          if (target != null) {
+            actions.open_file({
+              path: target,
+              foreground: !info.ctrl_down
+            });
+          }
+          break;
+        case "set":
+          actions.set_active_tab("settings");
+      }
+    },
+    [selected, actions]
+  );
+
+  const on_change = useDebounce(
+    React.useCallback(
+      (value: string): void => {
+        reset_cursor();
+        actions.setState({ search: value });
+      },
+      [reset_cursor, actions]
+    ),
+    50
```

Comment:
300ms -> 50ms

With WindowedList and the performance updates, 50ms is comfortable.